plugin/kubernetes/handler.go

```diff
@@ -30,29 +30,31 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 		err     error
 	)
 
-	switch state.Type() {
-	case "A":
+	switch state.QType() {
```

Comment:
Type() should just return AXFR  - I hope? Makes the PR smaller because it cuts out of the churn here.

Reply:
I thought it made sense to switch to using the constants for comparison versus a string comparison.

Reply:
yes, probably faster as well - should do this in diff. PR